src/parser/jobs/blm/modules/Procs.tsx

```diff
@@ -48,10 +34,14 @@ export default class Procs extends CoreProcs {
 		this.castTime.setInstantCastAdjustment([event.action], event.timestamp, event.timestamp)
 
 		// Thunder procs used while sharpcast is up re-grant the proc status without technically removing it, so we need to forcibly add the 'removal' here to keep the 'dropped' counting correct
-		if ((event.action === this.data.actions.THUNDER_III.id || event.action === this.data.actions.THUNDER_IV.id) && this.hasSharpcast) {
+		if ((event.action === this.data.actions.THUNDER_III.id || event.action === this.data.actions.THUNDER_IV.id) && this.actors.current.hasStatus(this.data.statuses.SHARPCAST.id)) {
 			this.tryAddEventToRemovals(procGroup, event)
+
+			// Make sure a Sharpcasted thunder always results in an open window, in case the log had weird status timing
+			if (!this.currentWindows.has(procGroup)) {
+				this.currentWindows.set(procGroup, {start: event.timestamp})
+			}
```

Comment:
It's a side effect of the "oops, the status went away earlier than it should have" event ordering, but in practice it's not going to matter much for BLM since it'll just end up looking to users (if they pay that close attention in the first place) like a DoT tic that procced right before the Sharpcast got used. It's better than the alternative of telling players they dropped a proc/Sharpcast buff when they actually did use them.